### PR TITLE
balance page - modal

### PR DESCRIPTION
**Pull request by krawczykkatarzyna**
Screen Shot 2023-11-21 at 09 50 36

### CODE REVIEW

File: src/components/frame/v5/pages/BalancePage/partials/BalanceTable.tsx [hunk 1]
```diff
@@ -14,9 +18,22 @@ const BalaceTable: FC<BalaceTableProps> = ({
...
+  const { balances, nativeToken, status, colonyAddress } = colony || {};
   const { nativeToken: nativeTokenStatus } = status || {};
-  const onAddFunds = () => {}; // @TODO: open modal
+
+  const {
+    avatarModalToggle: [
+      isAddFundsModalOpened,
+      { toggleOn: toggleAddFundsModalOn, toggleOff: toggleAddFundsModalOff },
+    ],
+  } = useActionSidebarContext();
```
Comment (joanna-pagepro) [inline src/components/frame/v5/pages/BalancePage/partials/BalanceTable.tsx:29]: Why are you using sidebar context here?

File: src/components/frame/v5/pages/BalancePage/partials/BalanceTable.tsx [hunk 2]
```diff
@@ -55,6 +72,28 @@ const BalaceTable: FC<BalaceTableProps> = ({
...
+        buttonMode="primarySolid"
+        icon="piggy-bank"
+      >
+        <h5 className="heading-5 mb-1.5">
+          {formatText({ id: 'balancePage.modal.title' })}
+        </h5>
+        <p className="text-md text-gray-600 mb-6">
+          {formatText({ id: 'balancePage.modal.subtitle' })}
+        </p>
+        <div className="flex sm:flex-row flex-col gap-4 sm:gap-2" />
```
Comment (joanna-pagepro) [inline src/components/frame/v5/pages/BalancePage/partials/BalanceTable.tsx:87]: I guess that this div is redundant here

File: src/components/frame/v5/pages/BalancePage/partials/BalanceTable.tsx [hunk 3]
```diff
@@ -55,6 +72,28 @@ const BalaceTable: FC<BalaceTableProps> = ({
...
+        <span className="flex text-1">
```
Comment (joanna-pagepro) [inline src/components/frame/v5/pages/BalancePage/partials/BalanceTable.tsx:88]: Please use other HTML tag here

File: src/components/v5/shared/CopyWallet/CopyWallet.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+import Button from '~v5/shared/Button';
+import { formatText } from '~utils/intl';
+import Icon from '~shared/Icon';
+import { CopyWalletProps } from './types';
+import { splitWalletAddress } from '~utils/splitWalletAddress';
+import { useMobile } from '~hooks';
+
+const displayName = 'v5.CopyWallet';
+
+const CopyWallet: FC<CopyWalletProps> = ({
```
Comment (joanna-pagepro) [inline src/components/v5/shared/CopyWallet/CopyWallet.tsx:12]: Please make this component reusable - don't use `walletAddress` here, just pass the value later, when you will be using it. Please also add it to storybook

File: src/components/v5/shared/CopyWallet/CopyWallet.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    <div className="bg-gray-50 rounded p-4 flex flex-col gap-4">
+      <div className="flex items-center gap-2">
+        <Icon name="cardholder" appearance={{ size: 'extraTiny' }} />
+        <span className="text-md">
+          {isMobile ? splitWalletAddress(walletAddress) : walletAddress}
+        </span>
+      </div>
+      {isCopied ? (
+        <Button mode="completed">
+          {formatText({ id: 'copy.addressCopied' })}
```
Comment (joanna-pagepro) [inline src/components/v5/shared/CopyWallet/CopyWallet.tsx:31]: Please add props to set the texts

File: src/components/v5/shared/CopyWallet/CopyWallet.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      ) : (
+        <Button
+          onClick={handleClipboardCopy}
+          onKeyDown={handleClipboardCopy}
+          disabled={disabled}
+          isFullSize
+          mode="primaryOutlineFull"
+          iconName="copy-simple"
+          className={clsx(
+            'flex items-center transition-all duration-normal text-sm text-gray-600 md:hover:text-blue-400',
```
Comment (joanna-pagepro) [inline src/components/v5/shared/CopyWallet/CopyWallet.tsx:42]: Please add `text-gray-600` when button is not disabled

**Review by joanna-pagepro — CHANGES_REQUESTED** [review]

File: src/components/v5/shared/CopyWallet/CopyWallet.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
```
Comment (KornelKwiatkowski) [inline src/components/v5/shared/CopyWallet/CopyWallet.tsx:43]: remove `text-gray-600` from default classes

**Review by KornelKwiatkowski — APPROVED** [review]

**Review by melyndav — CHANGES_REQUESTED** [review]
Hi @krawczykkatarzyna. Thanks for your work on this PR.

There are some UI fixes that are required.

1. The icon in the top left isn't the correct size as per Figma:
image


2. Size of the wallet icon next to the address is not centre aligned with the text to the right and is also not the correct size. Figma: 

image


3. The button has issues:

- not the correct Primary CTA button and has the wrong padding and text size
- Border colour is wrong 

image

4. When I cope the address the text size change, I think this is due to issue #3 above.

image


5. I don't see the add funds from wallet input and section?

image

**Comment by krawczykkatarzyna** [comment]
@melyndav 
1.  Can you check it again? From what I see size of the icon is the sam as on Figma design.
Screen Shot 2023-11-20 at 22 45 05
Screen Shot 2023-11-20 at 22 45 16
Screen Shot 2023-11-20 at 22 45 50
Screen Shot 2023-11-20 at 22 45 56

Other issues fixed.

6. As I spoke with @arrenv today he mentioned that API to create "add funds to wallet" is not ready yet  so we have to hide this section for now.

**Review by KornelKwiatkowski — DISMISSED** [review]

**Review by melyndav — CHANGES_REQUESTED** [review]
Hey @krawczykkatarzyna appreciate the updates. Everything looks good.

Can I request the title 'send funds to the colony' here just above the grey wallet & button be removed? Thank you!

image

**Comment by krawczykkatarzyna** [comment]
@melyndav title removed

**Review by KornelKwiatkowski — APPROVED** [review]

**Review by joanna-pagepro — APPROVED** [review]

**Review by melyndav — APPROVED** [review]
I appreciate that update @krawczykkatarzyna. Thank you!

**Review by arrenv — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**